src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,7 +1,87 @@
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
+import cn from 'classnames';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { ResponseError } from '../../types/ResponseError';
+import { MovieData } from '../../types/MovieData';
 import './FindMovie.scss';
+import { MovieCard } from '../MovieCard';
+
+const DEFAULT_POSTER
+  = 'https://via.placeholder.com/360x270.png?text=no%20preview';
+
+const DEFAULT_IMDB_URL = 'https://www.imdb.com/title/';
+
+type Props = {
+  addMovie: (film: Movie) => void;
+};
+
+export const FindMovie: React.FC<Props> = ({ addMovie }) => {
+  const [query, setQuery] = useState('');
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [data, setData] = useState<MovieData | null | ResponseError>(null);
+  const [foundMovie, setFoundMovie] = useState<Movie | null>(null);
+
+  const handleQueryChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setQuery(event.target.value);
+    if (isError) {
+      setIsError(false);
+    }
```

Comment:
yeah, the logic is correct, but it might look a bit confusing after the first look)
you can just set it false w/o checking: [article](https://pgarciacamou.medium.com/react-doesnt-always-trigger-a-re-render-on-setstate-4644212560a)